src/commonControls/listControl/ListControlAPL.ts

```diff
@@ -115,3 +116,234 @@ export namespace ListControlAPLPropsBuiltIns {
+                    },
+                ],
+                items: [
+                    {
+                        type: 'Sequence',
+                        scrollDirection: 'vertical',
+                        data: '${listItems}',
+                        width: '100%',
+                        height: '100%',
+                        paddingLeft: '0',
+                        numbered: true,
+                        items: [
+                            {
+                                type: 'Container',
+                                items: [
+                                    {
+                                        type: 'AlexaTextListItem',
+                                        touchForward: true,
+                                        hideOrdinal: false,
+                                        disabled: '${disableScreen}',
+                                        primaryText: '${data.primaryText}',
+                                        primaryAction: {
+                                            type: 'Sequential',
+                                            commands: [
+                                                {
+                                                    type: 'SendEvent',
+                                                    arguments: ['${controlId}', '${ordinal}'],
+                                                },
+                                                {
+                                                    type: 'SetValue',
+                                                    componentId: 'root',
+                                                    property: 'disableScreen',
+                                                    value: true,
+                                                },
+                                                {
+                                                    type: 'SetValue',
+                                                    componentId: 'root',
+                                                    property: 'debugText',
+                                                    value: 'Done Selected',
+                                                },
+                                            ],
+                                        },
+                                    },
+                                ],
+                            },
+                        ],
+                    },
+                ],
+            });
+
+            const itemIds: string[] = control.getListItemIDs(input);
+            // Create the inline document, which instantiates the Layout
+            const listItems = itemIds.map((x) => ({
+                primaryText: props.valueRenderer!(x, input).primaryText,
+            }));
+
+            return {
+                type: 'ListControl-touchForward',
+                controlId: control.id,
+                listItems,
+            };
+        } else if (props.renderStyle === 'imageList') {
```

Comment:
add TODO: offer customization of where the image is placed, highlight colors, etc.

design: is there some way we can make the row formatting most open.  e.g. developer provides a itemRenderer that produces a whole line item?

design: is there a good way to let developers create an entirely new rendering without having to fork the control.

Reply:
e.g. 

apl: {
/ **
  Function that renderers an APL component.

  Default: ListControlAPLRenderFuncs.TextList
  Builtins: see ListControlAPLRenderFuncs.*
  
**/
  renderFunc: ((state, input, resultBuilder)=> AplComponent   <-- giving this a type could help with comprehension & documentation.
}
User typically sets it to ListControlAPLRenderFuncs.ImageList
Pros:
 - only one prop
 - easy to drop in a custom func (and copy&paste from the builtins)
Cons:
 - builtins less discoverable?  Perhaps allow an enumerated-string type too for max discoverability... Doesn't seem valuable enough to do this right now.  we can add it later, or other mechanisms that have easier discoverability.

Reply:
Based on design discussions with @mikelid  and @sattpat 
- Introduced a new prop under apl -> renderComponent which developer can customize their own APL components or leverage builtin provided on each controls.
- Introduced new classes for each rendering Style with default and `.of(props)` method to render default APL components for control.
- All the classes are exported under APLComponenetBuiltIns.* namespace which can be used to customize the controls.

```ts
apl: {
      renderComponent: BuiltIns.ImageListRenderer.default
}
--------------------------------------------------------------------------------
apl: {
      renderComponent: BuiltIns.ImageListRenderer.of(props)           
}
--------------------------------------------------------------------------------
apl: {
      renderComponent: (controls, input, resultBuilder) => { [key:string]: any}
 }  
    
```